eng/generators.targets

```diff
@@ -43,10 +57,20 @@
 
   <!-- Use this complex item list based filtering to add the ProjectReference to make sure dotnet/runtime stays compatible with NuGet Static Graph Restore.
        That is required as the EnabledGenerators condition checks on the Reference and ProjectReference items and hence can't be a property condition. -->
+  <ItemGroup Condition="'@(EnabledGenerators)' != ''">
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime.InteropServices\gen\Microsoft.Interop.SourceGeneration\Microsoft.Interop.SourceGeneration.csproj"
+                      ReferenceOutputAssembly="false"
+                      OutputItemType="Analyzer" />
+  </ItemGroup>
   <ItemGroup Condition="'@(EnabledGenerators)' != '' and
                         @(EnabledGenerators->AnyHaveMetadataValue('Identity', 'LibraryImportGenerator'))">
```

Comment:
Did you consider applying the condition to the ProjectReference item directly so that you only need one item group?

Reply:
I wanted to follow the consistency with LibraryImportGenerator just so we keep the complexity in the same place for both. Hopefully we'll be able to simplify both of these soon.

Reply:
> Hopefully we'll be able to simplify both of these soon.

Just curious, how?

Reply:
Once we aren't building for .NET 6 or 7, I might be able to simplify things. Not sure though to what extent.